pragma/publisher/fetchers/index.py

```diff
@@ -0,0 +1,106 @@
+import asyncio
+import logging
+import time
+from typing import List, Union
+
+import requests
+from aiohttp import ClientSession
+
+from pragma.core.assets import PragmaAsset
+from pragma.core.entry import SpotEntry
+from pragma.publisher.types import PublisherFetchError, PublisherInterfaceT
+
+logger = logging.getLogger(__name__)
+
+
+import time
+from datetime import datetime
+from typing import List, Tuple, Union
+
+
+class AssetWeight:
+    def __init__(self, asset: PragmaAsset, weight: float):
+        self.asset = asset
+        self.weight = weight
+
+
+class IndexFetcher(PublisherInterfaceT):
+    fetcher: any
+    index_name: str
+    asset_weights: List[AssetWeight]
+
+    def __init__(
+        self,
+        fetcher: any,
+        index_name: str,
+        asset_weights: List[AssetWeight],
+    ):
+        self.fetcher = fetcher
+        self.index_name = index_name
+        self.asset_weights = asset_weights
+
+    async def fetch(
+        self, session: ClientSession
+    ) -> List[Union[SpotEntry, PublisherFetchError]]:
+        spot_entries = []
+        for asset_weight in self.asset_weights:
+            spot_entry = await self.fetcher._fetch_pair(asset_weight.asset, session)
+            if isinstance(spot_entry, PublisherFetchError):
+                return PublisherFetchError(
```

Comment:
```suggestion
                spot_entry = PublisherFetchError(
```

Reply:
By doing so, we still have to add a check to verify if the spot_entries does not contain a PublisherFetchError.Indeed, we do not want to compute the index value if there is a fetching error in the spot entries array. By returning directly the error, we make sure to handle the error properly and avoid the 'AttributeError: 'PublisherFetchError' object has no attribute 'price'' error instead

Reply:
To reproduce the same pattern as for the fetchers, we will need to implement maybe another class with a list of Index to work (analogy to the PRAGMA_ALL_ASSETS used by the fetchers). Currently the IndexFetcher work for a given index_name and List[assets, weights]